metadata-foundationdb/metadata-foundationdb.c

```diff
@@ -196,4 +198,35 @@ void virtual_disk_persist(FDBDatabase *database, VirtualDisk *virtual_disk)
 VirtualDisk *virtual_disk_get(FDBDatabase *database, ProtobufCBinaryData *key)
 {
     return (VirtualDisk *)proto_message_get(database, key->data, key->len, &virtual_disk__descriptor);
-}
\ No newline at end of file
+}
+
+void *run_net(void *_unused)
+{
+    chk(fdb_run_network());
+
+    return NULL;
+}
+
+void foundationdb_backend_initialize() {
+    fdb_select_api_version(FDB_API_VERSION);
+
+    chk(fdb_setup_network());
+
+    pthread_t net_thread;
+    assert(0 == pthread_create(&net_thread, NULL, run_net, NULL));
+
+    chk(fdb_create_database(NULL, &foundationdb_database));
+}
+
+MetadataBackend foundationdb_backend = {
+    .name = "foundationdb",
+    .initialize = foundationdb_backend_initialize,
+    .load = load_metadata,
+    .persist = metadata_persist
+};
```

Comment:
I think we also need a `foundationdb_backend_finalize` in order to destroy the database and stop the network thread.